OpenChange/MAPIStoreMailFolder.m

```diff
@@ -631,17 +628,22 @@ - (BOOL) synchroniseCache
           if (!lastModseq
               || ([lastModseq compare: modseq] == NSOrderedAscending))
             lastModseq = modseq;
+
+          if ([[result objectForKey: @"flags"] containsObject: @"deleted"])
+            [currentProperties setObject: changeNumber
+                                  forKey: @"SyncLastDeleteChangeNumber"];
```

Comment:
I think it will always be the latest one as `changeNumbers` array will have the elements stored in order, am I right? I don't see any case which the behaviour is different.